### ISSUE TITLE
Join endpoints are rate-limited even if the user has already joined the room

### ISSUE THREAD

**Issue by freelock** [matrix-org/synapse#8146]
### Description

In the Drupal integration, before sending a message into a room, the API first sends a join to make sure the account is present. Since 1.19.0, this is now resulting in a lot of M_LIMIT_EXCEEDED errors, and interrupting the rest of the processing if not caught/otherwise handled. My assumption is that this end point is idempotent, and by calling it before sending, I'm potentially saving a round trip of having to hit an endpoint to see if the account is already joined, and then again to join if it's not.

In #matrix-dev:matrix.org, @tulir pointed me to #8138 , so I updated the Docker image from the github develop.zip after that PR was merged (I think), but I'm still getting errors.

I think that #8008 is what caused this, and #8138 did not fix the ratelimit_overide user accounts. Needs an additional fix for this table?

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from _homeserver.log_, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.
-->

### Version information

<!-- IMPORTANT: please answer the following questions, to help us narrow down the problem -->

<!-- Was this issue identified on matrix.org or another homeserver? -->
- Homeserver: matrix.freelock.com

If not matrix.org:

<!--
 What version of Synapse is running?

You can find the Synapse version with this command:

$ curl http://localhost:8008/_synapse/admin/v1/server_version

(You may need to replace `localhost:8008` if Synapse is not configured to
listen on that port.)
-->
- 1.19.0:

- **Install method**:
<!-- examples: package manager/git clone/pip  -->
Docker image built from release tarball.

- **Platform**:
<!--
Tell us about the environment in which your homeserver is operating
distro, hardware, if it's running in a vm/container, etc.
-->
Debian stretch container.

**Comment by babolivier** [comment]
Thanks for your report! As I understanding, your issue isn't so much that the `ratelimit_override` table is being ignored on joins, but that the idempotency on the join endpoints broke as of #8008. This is a serious regression, which will be fixed by https://github.com/matrix-org/synapse/pull/8153 and released in 1.19.1 (coming out later this week).

**Comment by babolivier** [comment]
If the ratelimiter ignoring the override table (which is currently only used by the message ratelimiter and not the others) is also an issue for you, could you please open a separate issue on this repo?

**Comment by freelock** [comment]
Reading the PR, this sounds right to me -- the issue is I'm getting a ratelimit error on rooms the account is already a member of.

**Comment by babolivier** [comment]
https://github.com/matrix-org/synapse/pull/8153 has been merged into the release branch, so I'm closing this issue.

**Comment by babolivier** [comment]
FYI the fix to this issue has just been released in Synapse v1.19.1 :slightly_smiling_face:
